blocklylib-core/src/main/java/com/google/blockly/android/ui/fieldview/BasicFieldNumberView.java

```diff
@@ -163,6 +163,7 @@ protected void onFocusChanged(boolean focused, int direction, Rect previouslyFoc
                 // Replace empty string with value closest to zero.
                 mNumberField.setValue(0);
             }
+            updateLocalizedNumberFormatIfConstraintsChanged();
```

Comment:
How would the constraints change while the user is entering a number?

Reply:
It shouldn't by any user action, but it might by some app-specific event in another thread. This is more just about being safe, since the next line is using the localized number format.